src/components/SearchResultDetail/SearchResultDetail.tsx

```diff
@@ -4,16 +4,42 @@ import * as React from 'react';
 
 import ClipLoader from 'react-spinners/ClipLoader';
 
-import { findSearchResultAction } from '@/app/actions/action';
+import { findSearchResultAction, exportSearchResultPdfAction } from '@/app/actions/action';
 import { SearchResultStatus } from '@/constants/enum';
 import { useModal } from '@/hooks/useModal';
 import toast from '@/lib/toast';
 
 import SearchResultHtmlModal from './SearchResultHtmlModal';
 
-const SearchResultDetail = ({ id }: { id: string }) => {
+interface DownloadBase64Params {
+  base64String: string;
+  fileName: string;
+  mimeType: string;
+}
+
+const mimeTypePDF = 'application/pdf';
+
+const downloadBase64String = (downloadParams: DownloadBase64Params) => {
```

Comment:
The following method and prop names would be more descriptive without relying on too much implementation details:

```suggestion
const generateDownloadLink = (downloadLinkParams: DownloadLinkParams) => {
```